searchcore/src/vespa/searchcore/proton/matching/sameelementmodifier.h

```diff
@@ -22,7 +22,7 @@ class SameElementModifier : public search::query::TemplateTermVisitor<SameElemen
 
         vespalib::string prefix = n.getView() + ".";
         for (auto & child : n.getChildren()) {
-            search::query::StringBase * term  = static_cast<search::query::StringBase *>(child);
+            search::query::StringBase * term  = dynamic_cast<search::query::StringBase *>(child);
```

Comment:
please check term before using it
also, 'auto &' does not make it obvious that the input to the cast is a pointer, consider capturing as Node *

Reply:
Good point.
It did already a handle normal number terms as they are strings. 
However I introduced a extra TermNode parent so it would work fine with RangeTerms which where the other option you could get. 
Followup in #6075